### PR TITLE
Update the Illink Sample benchmark to support the PERFLAB_TARGET_FRAMEWORKS

### PR DESCRIPTION
**Pull request by LoopedBard3**
Update the Illink Sample real-world benchmark to support the PERFLAB_TARGET_FRAMEWORKS. Per https://github.com/dotnet/performance/pull/3511#issuecomment-1828709348.

### CODE REVIEW

**Review by kunalspathak — APPROVED** [review]

File: src/benchmarks/real-world/ILLink/Utilities.cs [hunk 1]
```diff
@@ -61,7 +61,7 @@ public static string PublishSampleProject(string projectFilePath, params string[
         {
             string outputDirectory = GenerateTempFolder();
             Directory.CreateDirectory(outputDirectory);
-            ProcessStartInfo processStartInfo = new ProcessStartInfo("dotnet", $"publish {projectFilePath} -r {CurrentRID} --self-contained -o {outputDirectory} {extraArgs.Aggregate("", (agg, val) => agg + " " + val)}");
+            ProcessStartInfo processStartInfo = new ProcessStartInfo("dotnet", $"publish {projectFilePath} -f {Environment.GetEnvironmentVariable("PERFLAB_TARGET_FRAMEWORKS") ?? "net9.0"} -r {CurrentRID} --self-contained -o {outputDirectory} {extraArgs.Aggregate("", (agg, val) => agg + " " + val)}");
```
Comment (cincuranet) [inline src/benchmarks/real-world/ILLink/Utilities.cs:64]: Will we forget about updating the `net9.0` here next year to keep it in sync? :)

**Review by cincuranet — DISMISSED** [review]

File: src/benchmarks/real-world/ILLink/Utilities.cs (hunk 1 above)
Comment (LoopedBard3) [reply src/benchmarks/real-world/ILLink/Utilities.cs:64]: Added a comment to the csproj change location to point out this update location. If you have other preferences please let me know!

Comment (cincuranet) [reply src/benchmarks/real-world/ILLink/Utilities.cs:64]: Excellent idea!

**Review by cincuranet — APPROVED** [review]

**Comment by LoopedBard3** [comment]
Failures match main, merging.

**Pull request merged**